muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -581,7 +617,28 @@ private boolean getCanSignup(WorkspaceEntity workspaceEntity) {
     }
   }
   
-  private CoursePickerWorkspace createRestModel(WorkspaceEntity workspaceEntity, String name, String nameExtension, String description, String educationTypeName, boolean canSignup, boolean isCourseMember) {
+  private boolean getIsAlreadyEvaluated(WorkspaceEntity workspaceEntity) {
+    if (sessionController.isLoggedIn()) {
+      WorkspaceUserEntity workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserByWorkspaceEntityAndUserIdentifier(workspaceEntity, sessionController.getLoggedUserEntity().defaultSchoolDataIdentifier());
+      Boolean isUnassessed = false;
+      if (workspaceUserEntity != null) {
+        WorkspaceRoleEntity workspaceRoleEntity = workspaceUserEntity.getWorkspaceUserRole();
+        WorkspaceRoleArchetype archetype = workspaceRoleEntity.getArchetype();
+        if (archetype.equals(WorkspaceRoleArchetype.STUDENT)) {
+          WorkspaceAssessmentState assessmentState = assessmentRequestController.getWorkspaceAssessmentState(workspaceUserEntity);
+          if (!assessmentState.getState().equals(WorkspaceAssessmentState.UNASSESSED)) {
```

Comment:
...so I guess something like...

`if (assessmentState().getState() == WorkspaceAssessmentState.PASS)`

We also have `WorkspaceAssessmentState.FAIL` but I think if someone has been evaluated with a non-passing grade, they are still allowed to try again, so let's settle with just `WorkspaceAssessmentState.PASS` for now.